lib/src/iterable_extensions.dart

```diff
@@ -601,6 +601,41 @@ extension IterableExtension<T> on Iterable<T> {
       yield slice;
     }
   }
+
+  /// The count of occurrences of each element.
+  ///
+  /// The map contains an entry for each unique element of this iterable,
+  /// as determined by `==`.
+  /// The value for each key is the number of times that element 
+  /// appears in the iterable.
+  ///
+  /// If there are elements that are equal (`==`), 
+  /// but not identical (`identical`), 
+  /// it is unspecified which of the elements is used as the key in the map. 
+  /// For example, if there are multiple lists with the same content, 
+  /// the map will only keep one of them as a key.
+  ///
+  /// Example:
+  /// ```dart
+  /// ['a', 'b', 'c', 'b', 'c', 'c'].frequencies; 
+  /// Returns: {'a': 1, 'b': 2, 'c': 3}.
+  /// ```
+  ///
+  /// Note: This method uses `==` to compare elements. 
+  /// For collections # `List`, `Set`, or `Map`, deep equality is not checked. 
```

Comment:
Not even shallow equality is checked, two lists are equal only if they are the same instance.

Reply:
I think this is "shallow" as it relates to `this`, not as it relates to the elements of `this` that are also collection typed.